napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1521 @@
+                                                   edge_color=edge_color,
+                                                   edge_width=1.5,
+                                                   symbol='square',
+                                                   scaling=False)
+                self._node._subvisuals[1].set_data(pos=None, width=0)
+        elif self._is_selecting:
+            box = create_box(self._drag_box)
+            edge_color = self._highlight_color
+            self._node._subvisuals[0].set_data(np.empty((0, 2)), size=0)
+            self._node._subvisuals[1].set_data(pos=box[[0, 2, 4, 6, 0]],
+                                               color=edge_color, width=1.5)
+        else:
+            self._node._subvisuals[0].set_data(np.empty((0, 2)), size=0)
+            self._node._subvisuals[1].set_data(pos=None, width=0)
+
+    def _finish_drawing(self):
+        """Resets properties involed in shape drawing so that a new shape can
+        then be drawn
+        """
+        index = copy(self._moving_shape)
+        self._is_moving = False
+        self.selected_shapes = []
+        self._drag_start = None
+        self._drag_box = None
+        self._fixed_vertex = None
+        self._moving_shape = None
+        self._moving_vertex = None
+        self._hover_shape = None
+        self._hover_vertex = None
+        if self._is_creating is True and self.mode == 'add_path':
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+            else:
+                self.data.edit(index, vertices[:-1])
+        if self._is_creating is True and self.mode == 'add_polygon':
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+        self._is_creating = False
+        self.refresh()
+
+    def remove_selected(self):
+        """Removes any selected shapes
+        """
+        to_remove = np.sort(self.selected_shapes)[::-1]
+        for index in to_remove:
+            self.data.remove(index)
+        self.selected_shapes = []
+        shape, vertex = self._shape_at(self._cursor_coord)
+        self._hover_shape = shape
+        self._hover_vertex = vertex
+        self.status = self.get_message(self._cursor_coord, shape, vertex)
+        self.refresh()
+
+    def _rotate_box(self, angle, center=[0, 0]):
+        """Perfroms a rotation on the selected box
+
+        Parameters
+        ----------
+        angle : float
+            angle specifying rotation of shapes in degrees.
+        center : list
+            coordinates of center of rotation.
+        """
+        theta = np.radians(angle)
+        transform = np.array([[np.cos(theta), np.sin(theta)],
+                             [-np.sin(theta), np.cos(theta)]])
+        box = self._selected_box - center
+        box = np.matmul(box, transform.T)
```

Comment:
```suggestion
        box = box @ transform.T
```

Reply:
fixed, I didn't know this notation!

Reply:
Oh, this is like the *best* thing from Python 3.5+! =) =) =)